pkg/controller/node/cidr_allocator.go

```diff
@@ -17,144 +17,171 @@ limitations under the License.
 package node
 
 import (
-	"encoding/binary"
 	"errors"
 	"fmt"
-	"math/big"
 	"net"
-	"sync"
+
+	"k8s.io/kubernetes/pkg/api"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/client/record"
+	"k8s.io/kubernetes/pkg/util/wait"
+
+	"github.com/golang/glog"
+)
+
+const (
+	// controls how many NodeSpec updates NC can process concurrently.
+	cidrUpdateWorkers   = 10
+	cidrUpdateQueueSize = 5000
```

Comment:
Weel - even if you moved it from somewhere, it makes sense to think why this is so. Or at least add a TODO to revisit this number.